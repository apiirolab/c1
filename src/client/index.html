<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>