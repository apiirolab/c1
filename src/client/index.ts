export const info = 'Client Info';
