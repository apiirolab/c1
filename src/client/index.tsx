import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
