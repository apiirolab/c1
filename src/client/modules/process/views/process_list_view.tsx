import * as React from 'react';

import { Header, List, ListProps, Message } from 'semantic-ui-react';
import styled, { StyledComponentClass } from 'styled-components';

import { ProcessListItemView } from './process_list_item_view';

const ProcessList: StyledComponentClass<ListProps, {}> = styled(List)`
  &&&&& .description {
    font-size: smaller;
  }
`;

type Props = {
  processes: any[];
};

let process: any;
let index: number;
export const ProcessListView: React.SFC<Props> = ({ processes }) => (
  <React.Fragment>
    <Header icon="find" content="Search" subheader="Find process resources" />
    <Choose>
      <When condition={processes && processes.length > 0}>
        <ProcessList divided relaxed>
          <For each="process" of={processes} index="index">
            <ProcessListItemView key={index} process={process} />
          </For>
        </ProcessList>
      </When>
      <Otherwise>
        <Message>There are no search results.</Message>
      </Otherwise>
    </Choose>
  </React.Fragment>
);

ProcessListView.displayName = 'ProcessListView';
