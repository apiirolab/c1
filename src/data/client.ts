import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
