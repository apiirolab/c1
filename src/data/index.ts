export const info = 'Data Info';
