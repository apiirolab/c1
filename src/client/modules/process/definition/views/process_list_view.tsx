import * as React from 'react';

import { Header, List, ListProps, Message } from 'semantic-ui-react';
import styled, { StyledComponentClass } from 'styled-components';

import { QueryTypes } from 'data/client';
import { ProcessItemView } from './process_item_view';

const ProcessList: StyledComponentClass<ListProps, {}> = styled(List)`
  max-width: 800px !important;

  &&&&& .description {
    font-size: smaller;
  }
`;

const ResultInfo = styled.div`
  color: #999;
`;

type Props = {
  processes: QueryTypes.BpmnProcessItem[];
  context: App.Context;
};

export const ProcessListView: React.SFC<Props> = ({ processes, context }) => (
  <React.Fragment>
    <ResultInfo>
      {processes.length === 1
        ? context.i18n`1 process available`
        : context.i18n`${processes.length.toString()} processes available`}
    </ResultInfo>
    <Header
      icon="tasks"
      content="Company Processes"
      subheader="Find and execute a new company process"
    />
    <Choose>
      <When condition={processes && processes.length > 0}>
        <ProcessList divided relaxed>
          {processes.map(process => (
            <ProcessItemView key={process.id} process={process} />
          ))}
        </ProcessList>
      </When>
      <Otherwise>
        <Message>No results</Message>
      </Otherwise>
    </Choose>
  </React.Fragment>
);

ProcessListView.displayName = 'ProcessListView';
