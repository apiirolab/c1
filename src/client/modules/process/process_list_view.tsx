import * as React from 'react';
import { Query, QueryResult } from 'react-apollo';

import { ProcessListView } from '../views/process_list_view';
import { ProcessListQuery } from './process_queries';

function renderResult<T extends QueryResult<any, any>>(
  result: T,
  component: JSX.Element
): JSX.Element {
  if (result.loading) {
    return <p>Loading...</p>;
  }
  if (result.error) {
    return <p>Error :(</p>;
  }
  return component;
}

export const ProcessListContainer = () => (
  <ProcessListQuery>
    {result => renderResult(result, <ProcessListView searchItems={result.data.processes} />)}
  </ProcessListQuery>
);
