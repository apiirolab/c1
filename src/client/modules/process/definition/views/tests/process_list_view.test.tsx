import * as React from 'react';

import { createProcesses } from 'client/modules/process/common/tests/process_test_data';
import { create, render } from 'client/tests';
import { QueryTypes } from 'data/client';
import { ProcessListView } from '../process_list_view';

import 'jest-styled-components';

describe('Process Definition', () => {
  describe('List', () => {
    describe('Empty', () => {
      const component = (
        <div>
          <ProcessListView processes={[]} context={create.context()} />
        </div>
      );

      it('renders empty', () => {
        const root = render(component);
        expect(root).toMatchSnapshot();
      });

      return { component };
    });

    describe('Records', () => {
      function componentWithData(processes: QueryTypes.BpmnProcessItem[] = createProcesses()) {
        return (
          <div>
            <ProcessListView processes={processes} context={create.context()} />
          </div>
        );
      }

      it('renders with data', () => {
        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      return { componentWithData };
    });
  });
});
