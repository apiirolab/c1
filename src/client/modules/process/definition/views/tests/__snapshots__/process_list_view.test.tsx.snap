// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Definition List Empty renders empty 1`] = `
<style>
.c0 {
  color: #999;
}</style>

<div>
  <div
    className="c0"
  >
    0 processes available
  </div>
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="tasks icon"
    />
    <div
      className="content"
    >
      Company Processes
      <div
        className="sub header"
      >
        Find and execute a new company process
      </div>
    </div>
  </div>
  <div
    className="ui message"
  >
    No results
  </div>
</div>
`;

exports[`Process Definition List Records renders with data 1`] = `
<style>
.c2.c2.c2.c2.c2 {
  vertical-align: top !important;
  width: 25px;
}

.c1 {
  max-width: 800px !important;
}

.c1.c1.c1.c1.c1 .description {
  font-size: smaller;
}

.c0 {
  color: #999;
}</style>

<div>
  <div
    className="c0"
  >
    4 processes available
  </div>
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="tasks icon"
    />
    <div
      className="content"
    >
      Company Processes
      <div
        className="sub header"
      >
        Find and execute a new company process
      </div>
    </div>
  </div>
  <div
    className="ui divided relaxed list c1"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="right floated content"
      >
        <button
          className="ui primary button"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="rocket icon"
          />
          Launch
        </button>
      </div>
      <i
        aria-hidden="true"
        className="briefcase icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/travel-process/preview/information/1"
        >
          Travel Process
        </a>
        <div
          className="description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="right floated content"
      >
        <button
          className="ui primary button"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="rocket icon"
          />
          Launch
        </button>
      </div>
      <i
        aria-hidden="true"
        className="money icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/purchase-process/preview/information/2"
        >
          Purchase Process
        </a>
        <div
          className="description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="right floated content"
      >
        <button
          className="ui primary button"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="rocket icon"
          />
          Launch
        </button>
      </div>
      <i
        aria-hidden="true"
        className="money bill alternate outline icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/sales-process/preview/information/3"
        >
          Sales Process
        </a>
        <div
          className="description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="right floated content"
      >
        <button
          className="ui primary button"
          onClick={[Function]}
          role="button"
        >
          <i
            aria-hidden="true"
            className="rocket icon"
          />
          Launch
        </button>
      </div>
      <i
        aria-hidden="true"
        className="user circle icon middle aligned c2"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/process/human-resource-process/preview/information/4"
        >
          Human Resource Process
        </a>
        <div
          className="description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
  </div>
</div>
`;
